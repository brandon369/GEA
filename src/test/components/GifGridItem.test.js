import React from "react";
import GifGridItem from "../../components/GifGridItem";
import {shallow} from "enzyme";

describe("Prueba en <GifGtidItem />", () => {
  const title = 'Un titulo';
  const url = '/';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  //Tomar un sanpchot delcomponente probar que coincida con el renderizado
  test("debe demostrar el componente correctamente", () => {

    expect(wrapper).toMatchSnapshot();

  })

  test('debede tener un pÃ¡rrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title)

  })

  test('debe de tener la imagen igual al  url y alt', () => {
    const img = wrapper.find('img')
    // console.log(img.html())
    // console.log(img.props().src)
    // console.log(img.prop('src'))

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })


  test('debee de tener la clase animate__fadeInDown', () => {
    const div = wrapper.find('div')
    expect(div.prop('className').includes('animate__fadeInDown')).toBe(true)
  })
})

