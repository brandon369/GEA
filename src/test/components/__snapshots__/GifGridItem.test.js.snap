// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGtidItem /> debe demostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un titulo"
    src="/"
  />
  <p>
    Un titulo
  </p>
</div>
`;
